### DISCUSSION TITLE
Config load/save for settings

### DISCUSSION THREAD

**Discussion by Woisek** [vladmandic/sdnext#125] [Ideas]
Inspired by the current circumstances, where due to a few hurdles a reinstallation was necessary and then you always have to make the settings again, it would be great if there was a way to save all settings and load them again later if necessary. Saved settings would also be very helpful when installing on another computer.

**Comment by TomerGadol** [comment]
To my understanding, you can save your config.json file that holds most, if not all, of your settings. This was you can also move it between PCs.

**Reply by Woisek** [reply]
I know about the config.json file. But AFAIK it can be re-written/changed by the system (while an update maybe?). Also, it seems that some values are fixed values, not saved when changing and then restarted. Like Sampler, sampling steps, CFG etc.
Furthermore, it would be much more convenient for the user to have a load/save (and maybe a "default") button in the settings than to search around in file directories and edit json files with the risk of making the file and the system unusable.
IMHO.

**Reply by TomerGadol** [reply]
The default sampler, steps, CFG, and other settings can be modified in ui-config.json.
I agree that a UI-based solution would be more convenient but at the moment, and considering the flood of new users to this repo, we need to make do with what we have.

**Reply by Woisek** [reply]
> The default sampler, steps, CFG, and other settings can be modified in ui-config.json. 

Yes. And that's why I have wrote:

> ... than to search around in file directories and edit json files with the risk of making the file and the system unusable.

Not every user is so tech savvy like most of "us", just because they use A1111/Automatic.
Or said differntly: Just because the User _uses_ A1111/Automatic, doesn't necessarly mean he _knows how_ all is working. Two pairs of shoes.

**Comment by vladmandic** [comment]
currently NONE of the UI settings are persistent. And only way to make them persistent is to edit `ui-config.json`
and ALL of system settings are persistent and can be edited via UI->Settings which modifies `config.json`

either or both files can be deleted at any time and then server will recreate default configuration.

I understand the ask to make UI settings persistent, but that would be a massive project as each UI variable would have to be touched to have a callback to save.

Only way for that to happen is if someone volunteers and implements that - I'd be for it. But I really don't have time given other items in the backlog.

**Reply by CapsAdmin** [reply]
https://github.com/ilian6806/stable-diffusion-webui-state works kind of okay for UI settings, not sure if it works in this repo though

**Reply by vladmandic** [reply]
it should work. but it only saves/loads some small subset of ui settings. and even those are hardcoded (you can only select from small set of available ones and if any of them change or get renamed, though luck). to do it right would mean doing what i described above.

**Reply by Woisek** [reply]
> It can be extended to preserve basically everything in the UI.

Maybe he is willing to expand it in the near future.
